src/basset.cpp

```diff
@@ -31,39 +31,40 @@ using std::literals::string_literals::operator""s;
 
 string json_escape(const string &str) {
   static const char hex[] = "0123456789abcdef";
+  char u[] = "u0000";
```

Comment:
Let's move this variable (and `hex` as well) to the `default:` block where it's used.

Reply:
Wouldn't `u[]` be initialized every time 'default:' is entered (while now its once per the string)?

Reply:
I think that compiler is clever enough to figure out the most efficient way to handle the variable, maybe the variable will not even be present in the assembly and bytes will be written directly into the final buffer.

My primary concern would be to limit scope of variables to make code easier to understand for humans as suggested in:
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-scope
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-cond
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es21-dont-introduce-a-variable-or-constant-before-you-need-to-use-it
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es26-dont-use-a-variable-for-two-unrelated-purposes